core/src/main/kotlin/org/evomaster/core/search/algorithms/AbstractGeneticAlgorithm.kt

```diff
@@ -141,13 +154,15 @@ abstract class AbstractGeneticAlgorithm<T> : SearchAlgorithm<T>() where T : Indi
      * Each element is generated by sampling and evaluated via the fitness function.
      * Stops early if the time budget is exceeded.
      */
-    private fun sampleSuite(): WtsEvalIndividual<T> {
+    protected fun sampleSuite(): WtsEvalIndividual<T> {
         val n = 1 + randomness.nextInt(config.maxSearchSuiteSize)
         val suite = WtsEvalIndividual<T>(mutableListOf())
 
         for (i in 1..n) {
             ff.calculateCoverage(sampler.sample(), modifiedSpec = null)?.run {
-                archive.addIfNeeded(this)
+                if (config.gaSolutionSource != EMConfig.GASolutionSource.POPULATION) {
```

Comment:
why this? @jgaleotti  did we explicitly speak about this? i don't remember... ie, whether we want to evaluated test-level archive for suite generation